bzt/modules/console.py

```diff
@@ -114,8 +114,9 @@ def prepare(self):
         if isinstance(self.engine.aggregator, ResultsProvider):
             self.engine.aggregator.add_listener(self)
 
-        self.disabled = self.settings.get("disable", False)
-        if self.disabled:
+        disable = str(self.settings.get('disable', 'auto')).lower()
+        if (disable == 'true') or ((disable == 'auto') and (not is_tty())):
```

Comment:
for case of "true" you'll get boolean value of option. This condition can be simplified into something like `self.disable=(disable==auto and not is_tty()) or disable`